scripts/apptester/apptester.py

```diff
@@ -251,33 +252,41 @@ def build_applications(self, name=None):
             build_result = [app_data, self.result_matrix[app], 'N/A']
             self.table_data.append(build_result)
 
-    def build_android_java_demo(self):
+
+    def execute_build(self, app_data, file_type):
+        try:
+            build_app = self.sandboxframe.build_demo(app_data['id'], file_type)
+            build_app_result = self.sandboxframe.is_build_successful(app_data['id'], file_type)
+
+            if build_app_result:
+                self.result_matrix[app_data['name']] = TestStatus.PASSED
+                print 'Building {}:\n{}'.format(app_data['name'], build_app)
+
+            else:
+                self.result_matrix[app_data['name']] = TestStatus.FAILED
+                print 'Building {}:\n{}'.format(app_data['name'], build_app)
+
+        except Exception as ex:
+            traceback.print_exc(ex)
+            self.result_matrix[app_data['name']] = TestStatus.FAILED
+
+    def build_java_android_demo(self):
         output = self.sandboxframe.get_demo_projects()
         for item in output:
             if 'java' in item['id'] or 'android' in item['id']:
                 destBinaryFile = item.get('destBinaryFile', None)
+
                 if not destBinaryFile:
                     continue
                 else:
-                    try:
-                        build_app = self.sandboxframe.build_android_java_demo(item['id'])
-                        build_app_result = self.sandboxframe.is_build_successful(item['id'])
-
-                        if build_app_result:
-                            self.result_matrix[item['name']] = TestStatus.PASSED
-                            print 'Building {}:\n{}'.format(item['name'], build_app)
-
-                        else:
-                            self.result_matrix[item['name']] = TestStatus.FAILED
-                            print 'Building {}:\n{}'.format(item['name'], build_app)
-                    
-                    except Exception as ex:
-                        traceback.print_exc(ex)
-                        self.result_matrix[item['name']] = TestStatus.FAILED
+                    self.execute_build(item, self.BINARY)
 
-        for app in self.result_matrix:
-            build_result = [app, self.result_matrix[app], 'N/A']
-            self.table_data.append(build_result)
+    def build_all_demo(self):
```

Comment:
This method actually does not build all of the demos. Based on code bellow, I would recommend to rename it to "check_src_archives" or "verify_src_archives"